tests/mock_tests/sfloworh_ut.cpp

```diff
@@ -0,0 +1,372 @@
+            portTable.set("PortInitDone", { { "lanes", "0" } });
+            gPortsOrch->addExistingData(&portTable);
+            static_cast<Orch*>(gPortsOrch)->doTask();
+        }
+
+        void initDb()
+        {
+            this->appDb = std::make_shared<swss::DBConnector>("APPL_DB", 0);
+            this->configDb = std::make_shared<swss::DBConnector>("CONFIG_DB", 0);
+            this->stateDb = std::make_shared<swss::DBConnector>("STATE_DB", 0);
+            this->chassisAppDb = std::make_shared<swss::DBConnector>("CHASSIS_APP_DB", 0);
+        }
+
+        std::shared_ptr<DBConnector> appDb;
+        std::shared_ptr<DBConnector> configDb;
+        std::shared_ptr<DBConnector> stateDb;
+        std::shared_ptr<DBConnector> chassisAppDb;
+
+        std::vector<Orch*> resourcesList;
+    };
+
+    /* Test enabling/disabling SFLOW */
+    TEST_F(SflowOrchTest, SflowEnableDisable)
+    {
+        MockSflowOrch mock_orch;
+        {
+            auto table1 = deque<KeyOpFieldsValuesTuple>(
+                {
+                    {
+                        "global",
+                        SET_COMMAND,
+                        {
+                            {"admin_state", "down"}
+                        }
+                    }
+                });
+            mock_orch.doSflowTableTask(table1);
+
+            ASSERT_FALSE(Portal::SflowOrchInternal::getSflowStatusEnable(mock_orch.get()));
+        }
+        {
+            auto table2 = deque<KeyOpFieldsValuesTuple>(
+                {
+                    {
+                        "global",
+                        SET_COMMAND,
+                        {
+                            {"admin_state", "up"}
+                        }
+                    }
+                });
+            mock_orch.doSflowTableTask(table2);
+
+            ASSERT_TRUE(Portal::SflowOrchInternal::getSflowStatusEnable(mock_orch.get()));
+        }
+    }
+
+    /* Test create/delete SFLOW */
+    TEST_F(SflowOrchTest, SflowCreateDelete)
+    {
+        MockSflowOrch mock_orch;
+        {
+            auto table3 = deque<KeyOpFieldsValuesTuple>(
+                {
+                    {
+                        "global",
+                        SET_COMMAND,
+                        {
+                            {"admin_state", "up"},
+                        }
+                    }
+                });
+            mock_orch.doSflowTableTask(table3);
+            ASSERT_TRUE(mock_orch.get().m_sflowStatus);
+        }
+        {
+            auto table4 = deque<KeyOpFieldsValuesTuple>(
+                {
+                    {
+                        "global",
+                        DEL_COMMAND,
+                        {
+                            {"admin_state", "up"},
+                        }
+                    }
+                });
+            mock_orch.doSflowTableTask(table4);
+            ASSERT_FALSE(mock_orch.get().m_sflowStatus);
+        }
+    }
+}
+
```

Comment:
Can you add a few cases for notifications coming from APP_SFLOW_SESSION_TABLE_NAME & APP_SFLOW_SAMPLE_RATE_TABLE_NAME tables?

Here is a reference on testing if sai api's are called as expected: https://github.com/Azure/sonic-swss/blob/master/tests/mock_tests/portsorch_ut.cpp#L92

Reply:
Not sure about the scope though 
But i agree, the test can certainly be improved later on demand.